packages/dataclass-editor/src/components/dataclass/data/dataclass-change-handlers.test.ts

```diff
@@ -0,0 +1,536 @@
+import type { Mock } from 'vitest';
+import { useAppContext } from '../../../context/AppContext';
+import type { DataClass, DataClassField, DataClassFieldEntity, DataClassFieldEntityAssociation, DataClassFieldModifier } from './dataclass';
+import { useDataClassChangeHandlers } from './dataclass-change-handlers';
+
+vi.mock('../../../context/AppContext', () => {
+  return {
+    useAppContext: vi.fn()
+  };
+});
+
+const setupAppContext = (dataClass: DataClass, selectedField?: number) => {
+  const setDataClass = vi.fn();
+  (useAppContext as Mock).mockReturnValue({
+    dataClass: dataClass,
+    setDataClass: setDataClass,
+    selectedField: selectedField
+  });
+  return setDataClass;
+};
```

Comment:
I know using mocks is better to avoid, but I don't know how to fix that. Any suggestions?

Reply:
why do you need to use a mock? you can also set a wrapper for the render or renderHook: https://github.com/axonivy/inscription-client/blob/master/packages/editor/src/test-utils/test-utils.tsx#L217-L226

Reply:
I also would only use mocks for hooks from external libraries, but I think it should not be necessary currently in your case.